besu/src/main/java/org/hyperledger/besu/controller/BesuController.java

```diff
@@ -211,6 +212,13 @@ BesuControllerBuilder fromGenesisConfig(
         throw new IllegalArgumentException("Unknown consensus mechanism defined");
       }
 
+      if (configOptions.getTerminalTotalDifficulty().isPresent()
+          && configOptions.getCheckpointOptions().isValid()
+          && (UInt256.fromHexString(configOptions.getCheckpointOptions().getTotalDifficulty().get())
+              .greaterOrEqualThan(configOptions.getTerminalTotalDifficulty().get()))) {
```

Comment:
Would be nice to extract this into a descriptively named method, e.g. checkpointIsLaterThanTerminalTotalDifficulty.

^ Writing that has made me wonder how a post-merge checkpoint will have a TD of >= TTD...won't all post-merge blocks have a TD = 0?

Reply:
Would be good to have a test for this too I think

Reply:
Every block after the merge has a 0 difficulty, but the total difficulty is the sum of every blocks difficulty up until the merge.
So every block after the merge has the same total difficulty which is basically 0xc70d815d562d3cfa955